test/hotspot/jtreg/serviceability/jvmti/vthread/StopThreadTest/StopThreadTest.java

```diff
@@ -168,49 +168,44 @@ static class TestTask implements Runnable {
         static Object lock = new Object();
         static void log(String str) { System.out.println(str); }
 
-        private volatile boolean started = false;
-        private volatile boolean finished = false;
+        static volatile boolean atPointA = false;
+        static volatile boolean finished = false;
 
-        static public void sleep(long millis) {
+        static void sleep(long millis) {
             try {
                 Thread.sleep(millis);
             } catch (InterruptedException e) {
                 throw new RuntimeException("Interruption in TestTask.sleep: \n\t" + e);
             }
         }
 
-        // Ensure thread is ready.
-        public void ensureStarted() {
-            while (!started) {
+        static void ensureAtPointA() {
+            while (!atPointA) {
                 sleep(1);
             }
         }
 
         // Ensure thread is finished.
-        public void ensureFinished() {
+        static void ensureFinished() {
             while (!finished) {
                 sleep(1);
             }
         }
 
         public void run() {
             log("TestTask.run: started");
-            started = true;
 
             boolean seenExceptionFromA = false;
             try {
                 A();
             } catch (AssertionError ex) {
                 log("TestTask.run: caught expected AssertionError from method A()");
                 seenExceptionFromA = true;
-                if (!Thread.currentThread().isVirtual()) { // platform thread
-                    // clear the interrupt status
-                    Thread.interrupted();
-                }
             }
             if (!seenExceptionFromA) {
                 StopThreadTest.setFailed("TestTask.run: expected AssertionError from method A()");
             }
+            Thread.interrupted();
```

Comment:
I'm not sure why this was move down this far. It seems you would want to do this before the `StopThreadTest.setFailed()` call just to be extra safe, so that means move it before the `if` statement.